packages/search/addon/services/navi-search-provider.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ * This service is used to discover all the available search providers.
+ */
+
+import Service from '@ember/service';
+import { getOwner } from '@ember/application';
+import config from 'ember-get-config';
+
+/* global requirejs */
+
+export default class NaviSearchProviderService extends Service {
+  /**
+   * @method getProvider
+   * @param name
+   * @returns {Object} search provider service object
+   */
+  getProvider(name) {
+    return getOwner(this).lookup(`service:navi-search/${name}`);
+  }
+
+  /**
+   * @method all
+   * @returns {Array} array of available search provider services
+   * @description Discovers all the search provider services
+   * under the folder services/navi-search/
+   */
+  all() {
+    const searchProvidersRegex = new RegExp(`^(?:${config.modulePrefix}/)?services/navi-search/([a-z-]*)$`),
+      searchProviderServices = Object.keys(requirejs.entries).filter(requirejsFileName =>
+        searchProvidersRegex.test(requirejsFileName)
+      );
+    return searchProviderServices.map(providerFileName =>
+      this.getProvider(searchProvidersRegex.exec(providerFileName)[1])
+    );
+  }
+
+  /**
+   * @method search
+   * @param {String} query
+   * @returns {Array} array of objects that contain the search results,
+   * the name of the result component as well as result ordering information
+   * @description uses all the discovered search providers to fetch search results
```

Comment:
perfect 👍

Reply:
We should probably have our jsdoc styles defined more clearly within navi. We usually put the description in the same line as the method name with a dash separating them rather than having its own tag and line.